File: 14. Longest Common Prefix/CodeVersion2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        l = 0
        pre = ""
        con = True
        
        while con:
            # We need to check if we are within bounds of all strings
            for i in range(len(strs)):
                if l >= len(strs[i]):
                    con = False
                    break
                if i == 0:
                    letter = strs[i][l]
                elif strs[i][l] != letter:
                    con = False
                    break
            if con:
                pre += letter
                l += 1

        return pre

File: 14. Longest Common Prefix/CodeVersion2.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) {
        return "";
    }

    let l = 0;
    let pre = "";
    let con = true;
    
    while (con) {
        for (let i = 0; i < strs.length; i++) {
            if (l >= strs[i].length) {
                con = false;
                break;
            }
            if (i === 0) {
                var letter = strs[i][l];
            } else if (strs[i][l] !== letter) {
                con = false;
                break;
            }
        }
        if (con) {
            pre += letter;
            l++;
        }
    }

    return pre;
    
};
